### PR TITLE
Fix deploy

### PR DESCRIPTION
**Pull request by tomrf1**
The deploy of `component-event-stream` was broken after the recent migration of the riffraff github action to use one role per repo.

The fix involved adding it here: https://github.com/guardian/riffraff-platform/pull/64
That PR also made the stack name consistent.

This PR migrates `component-event-stream` to use the github action instead of a script for the cloudformation updates.

### CODE REVIEW

File: .github/workflows/component-event-stream.yaml [hunk 1]
```diff
@@ -7,12 +7,13 @@ on:
     branches:
       - main
 jobs:
-  super_mode_calculator_build:
+  component_event_stream_build:
```
Comment (tomrf1) [inline .github/workflows/component-event-stream.yaml:10]: the name was wrong!

File: .github/workflows/component-event-stream.yaml [hunk 2]
```diff
@@ -7,12 +7,13 @@ on:
...
     if: github.repository_owner == 'guardian'
     # Required by actions-riff-raff
     permissions:
       id-token: write
       contents: read
+      pull-requests: write
```
Comment (tomrf1) [inline .github/workflows/component-event-stream.yaml:16]: now needed by `guardian/actions-riff-raff@v4`

File: .github/workflows/super-mode-calculator.yaml [hunk 3]
```diff
@@ -58,9 +58,8 @@ jobs:
 
       - uses: guardian/actions-riff-raff@v2
```
Comment (andrewHEguardian) [inline .github/workflows/super-mode-calculator.yaml:59]: worth updating to v4?

**Review by andrewHEguardian — APPROVED** [review]
great!

File: .github/workflows/super-mode-calculator.yaml (hunk 3 above)
Comment (tomrf1) [reply .github/workflows/super-mode-calculator.yaml:59]: yes but I think I'll save it for later :)

**Pull request merged**